src/polyfill/RTCDataChannel.ts

```diff
@@ -70,16 +73,17 @@ export default class RTCDataChannel extends EventTarget implements globalThis.RT
             this.dispatchEvent(new Event('bufferedamountlow'));
         });
 
-        this.#dataChannel.onMessage((data) => {
-            if (ArrayBuffer.isView(data)) {
-                if (this.binaryType == 'arraybuffer')
-                    data = data.buffer;
-                else
-                    data = Buffer.from(data.buffer);
+        this.#dataChannel.onMessage(message => {
+            let data: Blob | ArrayBufferLike | string
+            if (!ArrayBuffer.isView(message)) {
+                data = message
+            } else if (this.#binaryType === 'blob') {
+                data = new Blob([message])
```

Comment:
don't explicitly use buffer, also cast to Blob when the type requires it, this wasn't even done before